hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/DefaultNearCacheManager.java

```diff
@@ -120,18 +120,25 @@ public void clearAllNearCaches() {
 
     @Override
     public boolean destroyNearCache(String name) {
-        NearCache nearCache = nearCacheMap.remove(name);
-        if (nearCache != null) {
-            nearCache.destroy();
+        NearCache nearCache = nearCacheMap.get(name);
+        if (nearCache == null) {
+            synchronized (mutex) {
+                nearCache = nearCacheMap.get(name);
```

Comment:
Hmm, can't you replace this call with `nearCacheMap.remove(name);`?
This would save one interaction with the `nearCacheMap` and make the coed a bit more readable.